pcsx2/gui/FrameForGS.cpp

```diff
@@ -183,13 +185,19 @@ void GSPanel::DoResize()
 	else if( arr > 1 )
 		viewport.y = (int)( (double)viewport.y/arr + 0.5);
 
+	if (g_Conf->GSWindow.ScaleToImageSize && image_width > 0 && image_width < client.GetWidth() &&
+		image_height > 0 && image_height < client.GetHeight() && g_Conf->GSWindow.AspectRatio != AspectRatio_Stretch) {
+			viewport.x = image_width;
+			viewport.y = (int)((double)image_width / targetAr);
```

Comment:
I don't understand. I interpret this:
```
if the frame size is 640x480, then display a centered 640x480 frame
while leaving the rest black.
```
as meaning "avoid stretching the image and applying aspect ratios".